core/src/main/java/hudson/PluginManager.java

```diff
@@ -788,8 +788,6 @@ public boolean accept(File dir, String name) {
 
             LOGGER.log(INFO, "Upgraded Jenkins from version {0} to version {1}. Loaded detached plugins (and dependencies): {2}",
                     new Object[] {lastExecVersion, Jenkins.VERSION, loadedDetached});
-
-            InstallUtil.saveLastExecVersion();
```

Comment:
Can we keep this metadata in case it becomes useful later? Or was it just incorrect? Or redundant?

Reply:
It breaks the upgrade flow, it get's called in the Upgrade, Downgrade and Restart install state listeners so should always get called anyway

Reply:
Probably the cause of https://issues.jenkins.io/browse/JENKINS-65329

Reply:
well possibly (haven't investigated yet) but the fix isn't to re-add this.

The install state listeners should call this when completed